pyramid_restful/api.py

```diff
@@ -78,17 +80,29 @@ def factory(self, request, parent=None, name=None):
 
         :return     <pyramid_restful.services.AbstractService>
         """
+        traverse = request.matchdict['traverse']
+
         # show documentation at the root path
-        if not request.matchdict['traverse']:
+        if not traverse:
             return {}
         else:
-            name = name or request.matchdict['traverse'][0]
+            name = name or traverse[0]
 
             service = {}
             try:
                 service_type, service_object = self.services[name]
             except KeyError:
-                raise HTTPNotFound()
+                # look for direct pattern matches
+                traversed = '/' +   '/'.join(traverse)
```

Comment:
this will allow for custom subpath matching within the restful structure, ie:

```
from pyramid_restful import endpoint

@endpoint.get(pattern='/path/to/somewhere')
def somewhere(request):
    pass

@somewhere.endpoint.post()
def create_somewhere(request):
   pass
```

where this will now join together with the API route, but not be limited to the normal module/class/function bindings.